pkg/controllers/restore/controller.go

```diff
@@ -483,16 +483,26 @@ func customize(obj *unstructured.Unstructured) {
 		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
 	case "Cluster":
 		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
-		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+		switch obj.GetAPIVersion() {
+		case "fleet.cattle.io/v1alpha1":
 			// only warn error if field can't be found. In case there are breaking api changes error will be printed as Warn and user has workaround to patch it.
-			redeployAgentGeneration, _, err := unstructured.NestedInt64(obj.Object, "spec", "redeployAgentGeneration")
+			redeployAgentGeneration, _, err := unstructured.NestedFloat64(obj.Object, "spec", "redeployAgentGeneration")
```

Comment:
Note: I found out during testing that this has to be float64 due the json decoding.